lib/src/authentication/sign_in/sign_in_view_model.dart

```diff
@@ -1,22 +1,32 @@
 import "package:flutter/material.dart";
+import "package:flutter_gen/gen_l10n/app_localizations.dart";
 import "package:go_router/go_router.dart";
+import "package:kana_to_kanji/src/core/constants/authentication_method.dart";
+import "package:kana_to_kanji/src/core/repositories/user_repository.dart";
+import "package:kana_to_kanji/src/core/services/toaster_service.dart";
 import "package:kana_to_kanji/src/glossary/glossary_view.dart";
+import "package:kana_to_kanji/src/locator.dart";
 import "package:stacked/stacked.dart";
 
 class SignInViewModel extends BaseViewModel {
+  final UserRepository _userRepository = locator<UserRepository>();
+  final ToasterService _toasterService = locator<ToasterService>();
+
   final formKey = GlobalKey<FormState>();
   final TextEditingController emailController = TextEditingController();
   final TextEditingController passwordController = TextEditingController();
   final FocusScopeNode focusScopeNode = FocusScopeNode();
 
   final GoRouter router;
+  final BuildContext context;
+  final AppLocalizations l10n;
 
   bool get isSignInButtonEnabled =>
       emailController.value.text.isNotEmpty &&
       passwordController.text.isNotEmpty &&
       (formKey.currentState != null && formKey.currentState!.validate());
 
-  SignInViewModel(this.router);
+  SignInViewModel(this.context, this.router, this.l10n);
```

Comment:
We cannot pass the context here unfortunately because it can become "dirty" fairly quickly. Instead, it should be pass to the method that needs it directly

Reply:
Done